app/server/methods/data.js

```diff
@@ -17,10 +80,37 @@ Meteor.methods({
         residents: Residents.find().fetch(),
         roles: Meteor.roles.find().fetch(),
         settings: Settings.find().fetch(),
-        users: Meteor.users.find().fetch(),
-      }
+        users: Meteor.users.find().fetch()
+      };
 
       return exportData;
     }
+  },
+
+  "file-upload": async function(fileData) {
+    if (!fileData) return "No Data recieved!!";
+
+    let jsonData;
+    try {
+      jsonData = JSON.parse(fileData);
+
+      if (!typeof jsonData === "object") throw "JSON not in required format";
+
+      Object.keys(jsonData).forEach(Meteor.bindEnvironment(async collectionName => {
```

Comment:
See above for a simpler way to achieve this. I.e. we probably don't need the `bindEnvironment` or async stuff to achieve this goal.